File: Chapter 2 - Selection Sort/selection_sort.java
import java.util.*;

public class Main {

    public static int findSmallest(ArrayList<Integer> arr) {
        int smallest = arr.get(0);
        int smallestIndex = 0;

        for (int i = 1; i < arr.size(); i++) {
            if (arr.get(i) < smallest) {
                smallest = arr.get(i);
                smallestIndex = i;
            }
        }

        return smallestIndex;
    }

    public static ArrayList<Integer> selectionSort(ArrayList<Integer> arr) {
        ArrayList<Integer> newArr = new ArrayList<>();

        while (!arr.isEmpty()) {
            int smallestIndex = findSmallest(arr);

            newArr.add(arr.get(smallestIndex));
            arr.remove(smallestIndex);
        }

        return newArr;
    }

    public static void main(String[] args) {
        ArrayList<Integer> arr =
                new ArrayList<>(Arrays.asList(5, 3, 6, 2, 10));

        System.out.println(selectionSort(arr));
    }
}

File: Chapter 2 - Selection Sort/selection_sort.cpp
#include <iostream>
#include <vector>
using namespace std;

int findSmallest(vector<int>& arr) {
    int smallest = arr[0];
    int smallest_index = 0;

    for (int i = 1; i < arr.size(); i++) {
        if (arr[i] < smallest) {
            smallest = arr[i];
            smallest_index = i;
        }
    }

    return smallest_index;
}

vector<int> selectionSort(vector<int> arr) {
    vector<int> newArr;

    while (!arr.empty()) {
        int smallest_index = findSmallest(arr);

        newArr.push_back(arr[smallest_index]);
        arr.erase(arr.begin() + smallest_index);
    }

    return newArr;
}

int main() {
    vector<int> arr = {5, 3, 6, 2, 10};

    vector<int> sorted = selectionSort(arr);

    for (int num : sorted) {
        cout << num << " ";
    }

    return 0;
}